File: src/FactoryMethod/src/factories/serie-factory.ts
import { MidiaFactory } from './midia-factory';
import { Midia } from '../models/midia';
import { Serie } from '../models/serie';

export class SerieFactory implements MidiaFactory {
    public criarMidia(
        titulo: string, 
        descricao: string, 
        anoLancamento: number, 
        duracao: number, 
        genero: string, 
        classificacao: string, 
        elenco: string[]
    ): Midia {
        return new Serie(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco);
    }
}

File: src/FactoryMethod/src/SerieFactory.java
import java.util.ArrayList;

public class SerieFactory implements MidiaFactory {
    @Override
    public Midia criarMidia(String titulo, String descricao, int anoLancamento, int duracao, String genero, String classificacao, ArrayList<String> elenco) {
        return new Serie(titulo, descricao, anoLancamento, duracao, genero, classificacao, elenco);
    }
}
